src/Features/Core/Portable/ConvertAutoPropertyToFullProperty/AbstractConvertAutoPropertyToFullPropertyCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeGeneration;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Simplification;
+using static Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.SymbolSpecification;
+
+namespace Microsoft.CodeAnalysis.ConvertAutoPropertyToFullProperty
+{
+    internal abstract class AbstractConvertAutoPropertyToFullPropertyCodeRefactoringProvider
+        : CodeRefactoringProvider
+    {
+        internal abstract SyntaxNode GetProperty(SyntaxToken token);
+        internal abstract string GetUniqueName(string fieldName, IPropertySymbol property);
+        internal abstract SyntaxNode GetInitializerValue(SyntaxNode property);
+        internal abstract SyntaxNode GetPropertyWithoutInitializer(SyntaxNode property);
+        internal abstract (SyntaxNode newGetAccessor, SyntaxNode newSetAccessor) GetNewAccessors(
+            DocumentOptionSet options, SyntaxNode property, string fieldName, SyntaxGenerator generator);
+        internal abstract SyntaxNode GetTypeBlock(SyntaxNode syntaxNode);
+        internal abstract SyntaxNode ConvertPropertyToExpressionBodyIfDesired(DocumentOptionSet options, SyntaxNode fullProperty);
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(context.Span.Start);
```

Comment:
It'd be cool if we used the whole span and applied it to everything touched by the span. But this is fine until someone asks for it. :smile:

Reply:
See one of my top coments.  This will find the first token of a property if you're in the doc comment for it.   We should pass along the user position to GetProperty and ensure that it's really within the header region of the property.